### ISSUE TITLE
While hovering with the mouse data, netbox should offer "Copy to clipboard" box

### ISSUE THREAD

**Issue by Xemanth** [netbox-community/netbox#17229]
### NetBox version

v4.0.8

### Feature type

New functionality

### Proposed functionality

For example, in sites, pointing the mouse cursor to a record, like "physical address" should popup "copy to clipboard" box.

Similar as shown in the video here: https://gitlab.com/gitlab-org/gitlab-foss/-/issues/32490

### Use case

Marking the text is time-consuming, and it's easy to miss something from the record data.

### Database changes

I don't think this requires database changes.

**Comment by goteamkor** [comment]
this would actually be a pretty slick feature.

**Comment by Domoninic** [comment]
In the example given there is no 'copy to clipboard" button so where would this pop-up appear?
Pop-ups have a bad habit of appearing on top of the fields obscuring the data making it difficult move the mouse over any fields.

**Comment by goteamkor** [comment]
> In the example given there is no 'copy to clipboard" button so where would this pop-up appear? Pop-ups have a bad habit of appearing on top of the fields obscuring the data making it difficult move the mouse over any fields.

did you follow the link that shows the video?


https://github.com/user-attachments/assets/81291a15-880d-4123-9172-4485c97017d9

**Comment by Domoninic** [comment]
> > In the example given there is no 'copy to clipboard" button so where would this pop-up appear? Pop-ups have a bad habit of appearing on top of the fields obscuring the data making it difficult move the mouse over any fields.
> 
> did you follow the link that shows the video?
> hover-copy.mp4

Yes I did, I was referring to 'For example, in sites, pointing the mouse cursor to a record, like "physical address" should popup "copy to clipboard" box.'
 This contrasts with the linked video having a 'copy to clipboard' button on which to anchor the pop-up.

**Comment by ITJamie** [comment]
While a nice addition i would hope that this does not become a replacement for the copy to clipboard buttons in the longterm. Waiting for a hover event to trigger before being able to copy data can be fustrating

**Comment by goteamkor** [comment]
In the above video the button appears near instantaneously. If that exact copy to clipboard method were implemented that wouldn't be quick enough?

**Comment by Xemanth** [comment]
> In the above video the button appears near instantaneously. If that exact copy to clipboard method were implemented that wouldn't be quick enough?

I would be happy with that speed! 👌

**Comment by jeremystretch** [comment]
This should probably be implemented as a generic CSS class that can be attached to a containing element that will render a copy-to-clipboard button on hover.
